patch_recurrence/steps.py

```diff
@@ -60,25 +85,31 @@ def smoothen(scaled_imgs, patches, constants):
 
 def set_patch_buckets(patches, constants):
     """Assigning bucket numbers to each patch
+    Histogram equalization across every scaled image
     """
     num_buckets = constants.NUM_BUCKETS
     scaled_imgs = len(patches)
-    width = max(len(patches[i]) for i in range(scaled_imgs))
 
-    std_database = np.zeros((scaled_imgs, width))
     for k in range(scaled_imgs):
-        for index, patch in enumerate(patches[k]):
-            std_database[k][index] = patch.std_dev
+        std_database = []
+        patch_database = []
+        for patch in patches[k]:
+            std_database.append(patch.std_dev)
+            patch_database.append(0)
 
-    min_std = np.amin(std_database)
-    max_std = np.amax(std_database)
+        index = np.argsort(std_database)
+        interval = len(std_database) / num_buckets
 
-    norm_std = np.floor((std_database - min_std) * (num_buckets - 1) / (max_std - min_std))
-    for k in range(scaled_imgs):
-        map(lambda (i, x): setattr(x, 'bucket', norm_std[k][i]), enumerate(patches[k]))
+        for i in range(num_buckets):
```

Comment:
You might miss a few patches in this loop. I think this can be done in a cleaner way, but if it's not time consuming it's fine

Reply:
Takes less than a second. Why will I miss a few patches? Yes, I agree it can be done in a cleaner way. Will rewrite it.